qiskit/circuit/quantumcircuit.py

```diff
@@ -1997,72 +1997,102 @@ def depth(
         filter_function: Optional[callable] = lambda x: not getattr(
             x.operation, "_directive", False
         ),
+        *,
+        recurse: bool = False,
```

Comment:
Should `QuantumCircuit.count_ops` and `QuantumCircuit.size` be similarly updated?

Reply:
I can do, if you'd like it for this PR?  I was originally just targetting the parts that are necessary for the analysis passes to work, but this shouldn't be much additional work.

Reply:
We decided to remove the public `QuantumCircuit` changes from this release (`DAGCircuit` generally has less public visibility, so we're not as worried by that).